integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/CarbonAlterTableColRenameDataTypeChangeCommand.scala

```diff
@@ -0,0 +1,345 @@
+import org.apache.carbondata.core.metadata.schema.table.column.CarbonColumn
+import org.apache.carbondata.events.{AlterTableColRenameAndDataTypeChangePostEvent, AlterTableColRenameAndDataTypeChangePreEvent, OperationContext, OperationListenerBus}
+import org.apache.carbondata.format.{ColumnSchema, SchemaEvolutionEntry, TableInfo}
+import org.apache.carbondata.spark.util.DataTypeConverterUtil
+
+private[sql] case class CarbonAlterTableColRenameDataTypeChangeCommand(
+    alterTableColRenameAndDataTypeChangeModel: AlterTableColRenameAndDataTypeChangeModel)
+  extends MetadataCommand {
+
+  override def processMetadata(sparkSession: SparkSession): Seq[Row] = {
+    val LOGGER = LogServiceFactory.getLogService(this.getClass.getCanonicalName)
+    val tableName = alterTableColRenameAndDataTypeChangeModel.tableName
+    val dbName = alterTableColRenameAndDataTypeChangeModel.databaseName
+      .getOrElse(sparkSession.catalog.currentDatabase)
+    var isColumnRenameOnly = false
+    var isDataTypeChangeOnly = false
+    var isBothColRenameAndDataTypeChange = false
+    setAuditTable(dbName, tableName)
+    setAuditInfo(Map(
+      "column" -> alterTableColRenameAndDataTypeChangeModel.columnName,
+      "newColumn" -> alterTableColRenameAndDataTypeChangeModel.newColumnName,
+      "newType" -> alterTableColRenameAndDataTypeChangeModel.dataTypeInfo.dataType))
+    val locksToBeAcquired = List(LockUsage.METADATA_LOCK, LockUsage.COMPACTION_LOCK)
+    var locks = List.empty[ICarbonLock]
+    // get the latest carbon table and check for column existence
+    var carbonTable: CarbonTable = null
+    var timeStamp = 0L
+    try {
+      locks = AlterTableUtil
+        .validateTableAndAcquireLock(dbName, tableName, locksToBeAcquired)(sparkSession)
+      val metastore = CarbonEnv.getInstance(sparkSession).carbonMetastore
+      carbonTable = CarbonEnv.getCarbonTable(Some(dbName), tableName)(sparkSession)
+      if (!carbonTable.canAllow(carbonTable, TableOperation.ALTER_COL_RENAME_AND_CHANGE_DATATYPE,
+        alterTableColRenameAndDataTypeChangeModel.columnName)) {
+        throw new MalformedCarbonCommandException(
+          "alter table change datatype or column rename is not supported for index datamap")
+      }
+      val operationContext = new OperationContext
+      val alterTableColRenameAndDataTypeChangePreEvent =
+        AlterTableColRenameAndDataTypeChangePreEvent(sparkSession, carbonTable,
+          alterTableColRenameAndDataTypeChangeModel)
+      OperationListenerBus.getInstance()
+        .fireEvent(alterTableColRenameAndDataTypeChangePreEvent, operationContext)
+      val newColumnName = alterTableColRenameAndDataTypeChangeModel.newColumnName
+      val oldColumnName = alterTableColRenameAndDataTypeChangeModel.columnName
+      val carbonColumns = carbonTable.getCreateOrderColumn(tableName).asScala.filter(!_.isInvisible)
+      if (!carbonColumns.exists(_.getColName.equalsIgnoreCase(oldColumnName))) {
```

Comment:
Should it be judged that the new column name is not in the old columns?  if new column name is the same as one of old column, ti rhown exception